pkg/asset/internal/templates.go

```diff
@@ -299,24 +300,31 @@ spec:
         - --allocate-node-cidrs=true
         - --configure-cloud-routes=false
         - --cluster-cidr={{ .PodCIDR }}
-        - --root-ca-file=/etc/kubernetes/secrets/ca.crt
-        - --service-account-private-key-file=/etc/kubernetes/secrets/service-account.key
+        - --root-ca-file=/etc/secrets/ca.crt
+        - --service-account-private-key-file=/etc/secrets/service-account.key
         - --leader-elect=true
-        - --cloud-provider={{ .CloudProvider  }}
+        - --cloud-provider={{ .CloudProvider }}
+        - --kubeconfig=/etc/kubernetes/kubeconfig
```

Comment:
This shouldn't be needed -- the controller-manager will use the service account injected into the pod -- and shouldn't require access to host files.